src/TextField/TextField.jsx

```diff
@@ -306,7 +306,17 @@ const TextField = React.createClass({
   },
 
   componentWillMount() {
-    this._uniqueId = UniqueId.generate();
+    const {
+      hintText,
+      floatingLabelText,
+      id,
+    } = this.props;
+
+    warning(hintText || floatingLabelText || id, `We don't have enough information
```

Comment:
Okay, what do you think of adding `name` to the destructure props and use it in addition to the hintText/floatingLabelText that could be used to produce `uniqueId` to reduce chances of getting warning. I think name is often used and _generally_ unique across fields in a form (even if not enforced) and may be a better token to use.